comment/validators.py

```diff
@@ -0,0 +1,105 @@
+from django.contrib.contenttypes.models import ContentType
+from django.http import JsonResponse
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import status
+from rest_framework.exceptions import APIException
+
+from comment.models import Comment
+
+
+class CommentErrorValidation(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_detail = 'Bad Request'
+
+    def __init__(self, detail=None, status_code=None):
+        if status_code:
+            self.status_code = status_code
+        if not detail:
+            detail = self.default_detail
+        self.detail = detail
+
+
+class BaseValidatorMixin:
+    api = False
+    app_name = None
+    model_name = None
+    model_id = None
+    parent_id = None
+    error = None
+
+    def dispatch(self, request, *args, **kwargs):
+        """
+            let rest framework handle the exception to choose the right renderer
+            validate method shall be called in the derived API class
+        """
+        if self.api:
+            return super().dispatch(request, *args, **kwargs)
+        try:
+            self.validate(request)
+        except CommentErrorValidation as exc:
+            return JsonResponse({'type': 'error', 'detail': exc.detail}, status=400)
+        return super().dispatch(request, *args, **kwargs)
+
+    def validate(self, request):
+        self.app_name = request.GET.get("app_name") or request.POST.get("app_name")
+        self.model_name = request.GET.get("model_name") or request.POST.get("model_name")
+        self.model_id = request.GET.get("model_id") or request.POST.get("model_id")
+        self.parent_id = request.GET.get("parent_id") or request.POST.get("parent_id")
+
+
+class ContentTypeValidator(BaseValidatorMixin):
+    def validate(self, request):
+        super().validate(request)
+        if not self.model_name:
+            self.error = _("model name must be provided")
```

Comment:
Instead of calling the` gettext_lazy` at every error, the code could be kept cleaner by directly calling it only once, inside the class `CommentErrorValidation` when adding the `detail` attribute.

Also, shouldn't it's name be something like `CommentBadRequest`?

Reply:
Yes, why not. I got better name.